shell/browser/api/electron_api_app.cc

```diff
@@ -1280,6 +1280,9 @@ std::vector<gin_helper::Dictionary> App::GetAppMetrics(v8::Isolate* isolate) {
     std::optional<double> usage =
         process_metric.second->metrics->GetPlatformIndependentCPUUsage();
     cpu_dict.Set("percentCPUUsage", usage.value_or(0) / processor_count);
+    cpu_dict.Set(
+        "cumulativeCPUUsage",
+        process_metric.second->metrics->GetCumulativeCPUUsage().InSecondsF());
```

Comment:
[`GetPlatformIndependentCPUUsage()`](https://source.chromium.org/chromium/chromium/src/+/main:base/process/process_metrics.cc;l=126-131;drc=946e68c02af4db2e67476042dece1c08ab83ea11) calls `GetCumulativeCPUUsage` under the hood so it doesn't seem like this adds information. I'll admit it's a little strange that we divide by processor count above, but I'm not sure that oddity warrants a new API.

Can you say a little more about what you're expecting to achieve here?

Reply:
The division by the processor count isn't the point here; it's that the existing value goes through [this code](https://source.chromium.org/chromium/chromium/src/+/main:base/process/process_metrics.cc;l=104-124;drc=946e68c02af4db2e67476042dece1c08ab83ea11), which converts from cumulative usage to percentage usage _since the last call_. This means that multiple parts of a codebase calling `getAppMetrics` interfere with each other. Even if you had exactly one place in a codebase that called it, converting the results back to cumulative usage (to allow for e.g. determining the total usage during a long task, or average usage over its duration) would be finicky and require storing additional state.

Looking at the code you linked, I'm realizing that the exact implementation here has changed since the version of Chromium I was referencing when writing this; I think I should be able to avoid making two calls to `GetCumulativeCPUUsage` now. I'll make the change and push an improved version.

Reply:
Ah, understood. I'm surprised that that function is stateful given its name :S